### DISCUSSION TITLE
Adding onOperation outside of SubscriptionServer constructor

### DISCUSSION THREAD

**Discussion by findlay-hannam** [apollographql/apollo-server#5023] [Q&A]
I have an express server with the apollo middleware integration. I needed to integrate with a SSE API in the back-end, so I'm adding subscriptions.

I've set up subscriptions using [installSubscriptionHandlers](https://www.apollographql.com/docs/apollo-server/data/subscriptions/#using-with-middleware-integrations). When a client subscribes to the graphql, I want to use the variables to execute a function which will be used by the subscription.

I think [onOperation](https://www.apollographql.com/docs/graphql-subscriptions/lifecycle-events/) is the best way to go about this, but I can't see a way to register the hooks, or specify the url, while using `installSubscriptionHandlers`. Is this level of control possible? Or is there some other method I should be using?

**Comment by glasser** [comment]
Just to let you know, we are currently planning to remove the surface-level direct integration between `subscriptions-transport-ws` and `apollo-server` in AS v3; ie, `installSubscriptionHandlers` is going away. Given that `installSubscriptionHandlers` already seems to not support your use case, I'd suggest ignoring it and integrating `subscriptions-transport-ws` with your http server directly, or perhaps using `graphql-ws` which I've heard is more actively maintained.
